### PR TITLE
Task 8 - GET /api/articles

### PR DESCRIPTION
**Pull request by clam119**
For this PR I have done the following:

1. Added a function that responds with an array of all article objects that are sorted by their created_at date by a default descending order.
2. Should no query be passed into the request - it will return all article objects sorted by their created_at date by a default descending order.
3. Added query that will allow users to filter through the topics that they want to be returned - if a topic has no articles then it will return a 404 not found, and if a topic has article(s) then these will be returned to the user in a descending order by their created_at date.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -58,39 +57,39 @@ describe("Northcoders News API", () => {
...
-            article_id: 4,
-            comment_count: "0"
-          })
-        )
-      })
-    })
+        .get("/api/articles/4")
+        .expect(200)
+        .then(({ _body: { articleData } }) => {
+          const requestedArticle = articleData[0];
```
Comment (alexdevdavis) [inline __tests__/app.test.js:69]: not convinced that an array is an appropriate data type for a single article

File: __tests__/app.test.js [hunk 2]
```diff
@@ -58,39 +57,39 @@ describe("Northcoders News API", () => {
...
-            article_id: 1,
-            comment_count: "11"
-          })
-        )
-      })
-    })
+        .get("/api/articles/1")
+        .expect(200)
+        .then(({ _body: { articleData } }) => {
+          const requestedArticle = articleData[0];
```
Comment (alexdevdavis) [inline __tests__/app.test.js:84]: as per previous comment

File: __tests__/app.test.js [hunk 3]
```diff
@@ -120,85 +119,172 @@ describe("Northcoders News API", () => {
...
-        author: "butter_bridge",
-        body: "I find this existence challenging", 
-        created_at: "2020-07-09T20:11:00.000Z",
-        votes: 150
-      }
-      expect(updatedArticleData).toEqual(updatedArticle);
-    })
-  })
+  describe("PATCH /api/articles/:article_id", () => {
+    it("Status: 200 - Should respond with an article that has successfully had its voted increased to 150 from 100.", () => {
```
Comment (alexdevdavis) [inline __tests__/app.test.js]: typo

File: models/articles-model.js [hunk 4]
```diff
@@ -1,33 +1,93 @@
...
-    })
-}
+exports.fetchAllArticles = (topic) => {
+  const acceptableQueries = [];
 
+  let filteredQueryString = `SELECT articles.*, COUNT(comments.author) AS comment_count
+    FROM articles
+    LEFT JOIN comments
+    ON articles.article_id = comments.article_id
+    WHERE articles.topic = $1
```
Comment (alexdevdavis) [inline models/articles-model.js:32]: as this line is the only difference between the two, I suggest you have a look in the notes [here](https://notes.northcoders.com/courses/js-back-end/complex-queries) under 'optional WHERE clauses' to find a means of DRYing up your code

**Review by alexdevdavis — APPROVED** [review]
Your tests have a superb level of attention to detail: great work! 

- The comments I've added are refactor suggestions, to give your code more of a professional vibe; I strongly advise you to act on them, but you needn't submit another PR in order to proceed. :)

**Comment by clam119** [comment]
Thank you for the code review and kind words Alex! 😁 Just made the changes and now I understand why it's excessive to have a single article's data come back as an array 😎

**Pull request merged**